include/alp/ops.hpp

```diff
@@ -34,488 +35,102 @@ namespace alp {
 	 */
 	namespace operators {
 
-		/**
-		 * This operator discards all right-hand side input and simply copies the
-		 * left-hand side input to the output variable. It exposes the complete
-		 * interface detailed in alp::operators::internal::Operator. This operator
-		 * can be passed to any GraphBLAS function or object constructor.
-		 *
-		 * Mathematical notation: \f$ \odot(x,y)\ \to\ x \f$.
-		 *
-		 * \note A proper GraphBLAS program never uses the interface exposed by this
-		 *       operator directly, and instead simply passes the operator on to
-		 *       GraphBLAS functions.
-		 *
-		 * @tparam D1 The left-hand side input domain.
-		 * @tparam D2 The right-hand side input domain.
-		 * @tparam D3 The output domain.
-		 */
+		/** @see grb::operators::left_assign */
 		template< typename D1, typename D2 = D1, typename D3 = D2, enum Backend implementation = config::default_backend >
-		class left_assign : public internal::Operator< internal::left_assign< D1, D2, D3, implementation > > {
-		public:
-			template< typename A, typename B, typename C, enum Backend D >
-			using GenericOperator = left_assign< A, B, C, D >;
-			left_assign() {}
-		};
-
-		/** TODO documentation. */
+		using left_assign = grb::operators::left_assign< D1, D2, D3, grb::Backend::reference >;
```

Comment:
here and similar situations below - see question about backend in `base/internalops.hpp`.